Oregon Nexus - Freshman On Track PoweBI Dashboard/MSSQL/002-Config-Views.sql

```diff
@@ -0,0 +1,149 @@
+CREATE VIEW [dbo].[OnTrackReport]
+AS
+
+WITH
+
+/*
+
+	GRADE LEVELS
+
+*/
+
+GRADE_LEVELS
+AS
+(
+SELECT [Description]
+
+,CASE WHEN Description = 'Preschool/Prekindergarten'  THEN 'PK'
+	  WHEN Description like 'Kinder%'  THEN 'K'
+	  WHEN Description like 'First%'  THEN '1'
+	  WHEN Description like 'Second%'  THEN '2'
+	  WHEN Description like 'Third%'  THEN '3'
+	  WHEN Description like 'Fourth%'  THEN '4'
+	  WHEN Description like 'Fifth%'  THEN '5'
+	  WHEN Description like 'Six%'  THEN '6'
+	  WHEN Description like 'Seven%'  THEN '7'
+	  WHEN Description like 'Eight%'  THEN '8'
+	  WHEN Description like 'Ninth%'  THEN '9'
+	  WHEN Description like 'Ten%'  THEN '10'
+	  WHEN Description like 'Eleven%'  THEN '11'
+	  WHEN Description like 'Twel%'  THEN '12'
+	  WHEN Description = 'Grade 13'  THEN '13'
+ END as Grade_Num
+ FROM  [edfi].[Descriptor]
+ JOIN [edfi].[GradeLevelDescriptor]
+	ON [edfi].[GradeLevelDescriptor].GradeLevelDescriptorId = [edfi].[Descriptor].DescriptorId
+)
+
+/*
+	
+	RACE
+
+*/
+
+	,STUDENT_RACE
+	AS
+	(
+		SELECT sr.[StudentUSI]		as StudentID
+			,sr.[RaceDescriptorId]		as TypeID
+			,d.Description			as Description
+		FROM [edfi].[StudentEducationOrganizationAssociationRace] sr
+		JOIN [edfi].[Descriptor] d
+			ON sr.RaceDescriptorId = d.DescriptorId
+		JOIN [edfi].[RaceDescriptor]
+			ON [edfi].[RaceDescriptor].RaceDescriptorId = d.DescriptorId
+	)
+	 
+	
+	SELECT distinct
+      sd.StudentKey
+      ,stu.StudentUniqueId as SSID
+      ,sd.[StudentFirstName]
+      ,sd.[StudentMiddleName]
+      ,sd.[StudentLastName]
+	  ,gl.Grade_Num
+	  ,sd.SchoolKey
+	  ,sch.SchoolName
+      ,sd.[LimitedEnglishProficiency]
+      --,sd.[IsEconomicallyDisadvantaged]
+      ,sd.[IsHispanic]
+
+	  ,CASE WHEN sd.IsHispanic = '1' THEN 'Hispanic'
+	      WHEN (SELECT COUNT(StudentID) FROM STUDENT_RACE where StudentID = stu.StudentUSI ) > 1 THEN 'Multiple'
+	      ELSE (SELECT Description FROM STUDENT_RACE where StudentID = stu.StudentUSI )
+      END		as Race
+
+      ,sd.[Sex]
+	  ,gpa.GPAScore
+	  ,gpa.GPAIndicator				as gpa_Indicator
+	  ,marks.CountD					as marks_CountD
+	  ,marks.CountF					as marks_CountF
+	  ,marks.Indicator				as marks_Indicator
+	  ,atnd.AttendanceRate			as atnd_Rate
+	  ,atnd.AttendanceIndicator		as atnd_Indicator
+	  ,bhv.StateOffenses			as bhv_Offenses
+	  ,bhv.BehaviorIndicator		as bhv_Indicator
+
+	  ,CASE WHEN (gpa.GPAIndicator like 'Off%'  OR  marks.Indicator like 'Off%'  OR  atnd.AttendanceIndicator like 'Off%'  OR  bhv.BehaviorIndicator like 'Off%') THEN 'Off-Track'
+	        WHEN (gpa.GPAIndicator like 'Vul%'  OR  marks.Indicator like 'Vul%'  OR  atnd.AttendanceIndicator like 'Vul%'  OR  bhv.BehaviorIndicator like 'Vul%') THEN 'Vulnerable'
+			ELSE  'On-Track'
+		END  as Overall_Indicator,
+		rank() OVER(PARTITION BY sd.StudentKey ORDER BY AttendanceRate) RNK
+  FROM [analytics].[StudentSchoolDim] sd				
```

Comment:
Take into account that if a student is enrolled in more than one school, StudentSchoolDim will return the student more than once.